server/src/authServer.ts

```diff
@@ -30,6 +31,30 @@ app.get('/posts', authenticateToken, (req: Request, res: Response) => {
   res.json(posts.filter(post => post.username === user.name))
 })
 
+app.post('/register', (req: Request, res: Response) => {
+  const { username, password }: { username: string; password: string } = req.body
+
+  bcrypt
+    .hash(password, 10)
```

Comment:
Good job hashing the password! We need to salt it as well. Here's an [awesome video](https://www.youtube.com/watch?v=--tnZMuoK3E) explaining what a password hash, salt, and pepper each are.

This [StackOverflow post](https://stackoverflow.com/questions/43092071/how-should-i-store-salts-and-passwords-in-mongodb) shows a very basic example of how to use password salts. Notice how the password never leaves the backend.

Reply:
awesome thanks for the suggestion!